File: 500-keyboard-row/keyboard-row.cpp
#include <vector>
#include <string>
using namespace std;

class Solution {
public:
    bool isSame(string word, string row) {
        for (char ch : word) {
            ch = tolower(ch);
            if (row.find(ch) == -1) {
                return false;
            }
        }
        return true;
    }

    vector<string> findWords(vector<string>& words) {
        string first = "qwertyuiop";
        string second = "asdfghjkl";
        string third = "zxcvbnm";
        vector<string> ans;

        for (string word : words) {
            if (isSame(word, first) || isSame(word, second) || isSame(word, third)) {
                ans.push_back(word);
            }
        }

        return ans;
    }
};

File: 500-keyboard-row/keyboard-row.py
class Solution:
    def findWords(self, words: List[str]) -> List[str]:

        dic = {1:"eiopqrtuwy",2:"adfghjkls",3:"bcmnvxz"}
        res = []
        ans = []

        for i in words:
            res.append(i.lower())
        
        for i in range(len(res)):
            if set(res[i]).issubset(set(dic[1])):
                ans.append(words[i])

            elif set(res[i]).issubset(set(dic[2])):
                ans.append(words[i])

            elif set(res[i]).issubset(set(dic[3])):
                ans.append(words[i])

        return ans
        